File: test-fed.js
import fetch from 'node-fetch';

const BASE_URL = 'http://localhost:5000';

async function testFedAPIs() {
  console.log('🧪 Testing FED APIs...\n');

  try {
    // Test health endpoint
    console.log('🔍 Testing health endpoint...');
    const healthResponse = await fetch(`${BASE_URL}/api/health`);
    const healthData = await healthResponse.json();
    console.log('✅ Health:', healthData);
    console.log('');

    // Force update FED data
    console.log('🔄 Force updating FED data...');
    const updateResponse = await fetch(`${BASE_URL}/api/update/fed`, {
      method: 'POST'
    });
    const updateData = await updateResponse.json();
    console.log('✅ Update result:', updateData);
    console.log('');

    // Test FED updates endpoint
    console.log('🔍 Testing FED updates endpoint...');
    const fedResponse = await fetch(`${BASE_URL}/api/fed-updates`);
    const fedData = await fedResponse.json();
    console.log('✅ FED updates:', fedData);
    console.log(`📊 Found ${fedData.length} FED updates`);
    console.log('');

    // Test FRED indicators
    console.log('🔍 Testing FRED indicators...');
    const fredResponse = await fetch(`${BASE_URL}/api/fred/indicators`);
    const fredData = await fredResponse.json();
    console.log('✅ FRED indicators:', fredData);
    console.log('');

    // Test FRED rate history
    console.log('🔍 Testing FRED rate history...');
    const historyResponse = await fetch(`${BASE_URL}/api/fred/rate-history`);
    const historyData = await historyResponse.json();
    console.log('✅ FRED rate history:', historyData);
    console.log('');

  } catch (error) {
    console.error('❌ Error testing FED APIs:', error);
  }
}

testFedAPIs(); 

File: backend/tests/test_fed.py
#!/usr/bin/env python3
"""
Test script for Federal Reserve (FED) APIs
Equivalent to the original test-fed.js
"""

import asyncio
import httpx
import json
from datetime import datetime


BASE_URL = "http://localhost:5000"


async def test_fed_apis():
    """Test FED APIs"""
    print("🧪 Testing FED APIs...\n")

    async with httpx.AsyncClient() as client:
        try:
            # Test health endpoint
            print("🔍 Testing health endpoint...")
            health_response = await client.get(f"{BASE_URL}/api/health")
            health_data = health_response.json()
            print("✅ Health:", health_data)
            print("")

            # Test FED indicators
            print("🔍 Testing FRED indicators...")
            fred_response = await client.get(f"{BASE_URL}/api/fed/indicators")
            fred_data = fred_response.json()
            print("✅ FRED indicators:", fred_data)
            print("")

            # Test FRED rate
            print("🔍 Testing FRED rate...")
            rate_response = await client.get(f"{BASE_URL}/api/fed/rate")
            rate_data = rate_response.json()
            print("✅ FRED rate:", rate_data)
            print("")

            # Test FRED rate history
            print("🔍 Testing FRED rate history...")
            history_response = await client.get(f"{BASE_URL}/api/fed/rate-history")
            history_data = history_response.json()
            print("✅ FRED rate history:", history_data)
            print("")

            # Test whale transactions
            print("🔍 Testing whale transactions...")
            whale_response = await client.get(f"{BASE_URL}/api/whale-transactions")
            whale_data = whale_response.json()
            print("✅ Whale transactions:", whale_data)
            print(f"📊 Found {len(whale_data.get('data', []))} whale transactions")
            print("")

            # Test update whale transactions
            print("🔄 Testing update whale transactions...")
            update_whale_response = await client.post(f"{BASE_URL}/api/update/whale")
            update_whale_data = update_whale_response.json()
            print("✅ Update whale transactions:", update_whale_data)
            print("")

        except Exception as error:
            print(f"❌ Error in FED API tests: {error}")

    print("🏁 FED API Tests Completed!")


if __name__ == "__main__":
    asyncio.run(test_fed_apis())